### PR TITLE
Add more integration tests

### PR DESCRIPTION
**Pull request by lauralt**
## Reason for This PR

When switching from firecracker/versionize to this repo, this commit: https://github.com/firecracker-microvm/firecracker/commit/cf925ebf3292d6759c7145f9c67a6866023aea3d was missed.

## Description of Changes

Added versionize changes from that commit; also removed some weird checks made in union tests (Need to sync up with @iulianbarbu and @ioanachirca if it seems better now or further changes should be made here or in subsequent PRs).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

**Comment by acatangiu** [comment]
Is this PR a cherry-pick of https://github.com/firecracker-microvm/firecracker/commit/cf925ebf3292d6759c7145f9c67a6866023aea3d, or does it have any extra content?

**Comment by lauralt** [comment]
It doesn't have extra content (just some superfluous / wrong checks regarding unions testing removed). I will take a look why the coverage decreased even though the number of covered lines should be the same.

**Comment by lauralt** [comment]
Now the coverage should be fixed, I added one more test for enums (`test_versionize_enum()`) as it seems the enum versionizing was not completely covered by the other tests that were added in this PR. Coverage increased on x86 with 0.2, but on arm it decreased with 0.1 which seems pretty weird :-?.

File: tests/test.rs [hunk 1]
```diff
@@ -689,6 +677,1285 @@ fn test_versionize_union_with_array() {
...
+impl SomeStruct {
+    fn ser_u16(&mut self, target_version: u16) -> VersionizeResult<()> {
+        // Fail if semantic serialization is called for the latest version.
+        assert!(target_version < 2);
+        self.message.as_mut_fam_struct().padding += 2;
+        Ok(())
+    }
+}
+
+// TODO: remove this test once FamStructWrapper `Clone` implementation gets fixed.
```
Comment (ioanachirca) [inline tests/test.rs:1909]: nit: would be nice to add issue link

**Pull request merged**